Add `useDepositModal` hook

danieledamiani opened the pull request.

ShawTim commented:
i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?

danieledamiani commented:
> i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?

Yes, I've renamed it to `useDepositModalProps`

ShawTim commented:
> > i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?
> 
> Yes, I've renamed it to `useDepositModalProps`

variable name has changed but the hook name is still `useDepositModal`

danieledamiani commented on packages/tempus-client_v3/src/hooks/useDepositModal.ts:
```diff
@@ -0,0 +1,47 @@
+  usdRates: Map<Ticker, Decimal>;
+}
+
+export const [tempusPoolsForDepositModal$, setTempusPoolsForDepositModal] = createSignal<TempusPool[]>();
```
I've tried different ways but none that worked well

ShawTim replied:
maybe accept `TempusPool[]` as params of `useDepositModalProp()`?

danieledamiani replied:
it doesn't seem to work...

danieledamiani commented:
> > > i's thinking `useDepositModal` would help to create a `DepositModal` but it's not that case. maybe rename it as sth like `useDepositModalData` or sth else?
> > 
> > 
> > Yes, I've renamed it to `useDepositModalProps`
> 
> variable name has changed but the hook name is still `useDepositModal`

Renamed the hook

iamvukasin commented on packages/tempus-client_v3/src/config/ConfigManager.ts:
```diff
@@ -88,6 +114,41 @@ class ConfigManager {
+    const todayTime = today.getTime();
+
+    const filteredPools = this.getPoolList()
+      .filter(pool => {
+        if (filterByChain) {
+          return pool.chain === filterByChain;
+        }
+
+        return true;
+      })
+      .filter(pool => {
+        if (filterByToken) {
+          return pool.backingToken === filterByToken;
+        }
+
+        return true;
+      })
+      .filter(pool => {
+        if (filterByProtocol) {
+          return pool.protocol === filterByProtocol;
+        }
+
+        return true;
+      })
```
Can you replace this with `getFilteredPoolList`?

iamvukasin commented on packages/tempus-client_v3/src/hooks/useDepositModalData.ts:
```diff
@@ -0,0 +1,50 @@
+  filter(tempusPools => tempusPools.length > 0),
+  map(tempusPools => ({
+    tempusPools,
+    obsMaturityTerms: tempusPools.map(tempusPool => maturityTerm$(tempusPool)),
```
What does `obsMaturityTerms` mean?

danieledamiani replied:
`observable of MaturityTerms`
